mulighetsrommet-api/src/test/kotlin/no/nav/mulighetsrommet/api/validation/ValidationDslTest.kt

```diff
@@ -0,0 +1,27 @@
+package no.nav.mulighetsrommet.api.validation
+
+import io.kotest.assertions.arrow.core.shouldBeLeft
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.should
+import no.nav.mulighetsrommet.api.responses.FieldError
+import kotlin.contracts.ExperimentalContracts
+
+@OptIn(ExperimentalContracts::class)
+class ValidationDslTest : FunSpec({
+    test("poc") {
+        val a = validation {
+            val g: Int? = null
+
+            requireValid(g != null) {
+                FieldError.root("asdf")
```

Comment:
Liten kommentar til redundante constructors av FieldError som vi kanskje kan kvitte oss med:

- `FieldError.ofPointer(pointer, detail)` kan erstattes med å kalle `FieldError(pointer, detail)` i stedet
- `FieldError.root(detail)` kan erstattes med å kalle `FieldError.of(detail)`